src/Inciter/Partitioner.h

```diff
@@ -1283,7 +1284,7 @@ class Partitioner : public CBase_Partitioner< HostProxy,
     //! Signal back to host that we are ready for partitioning the mesh
     void signal2host_setup_complete( const CProxy_Transporter& host ) {
       Group::contribute(
-        CkCallback(CkIndex_Transporter::redn_wrapper_partition(NULL), host ));
+        CkCallback(CkIndex_Transporter::redn_wrapper_part(NULL), host ));
```

Comment:
Okay, thanks for the clarification.

This call signals the host that the mesh graph has been read and the cell centroids have been computed by the given PE. This is necessary but not sufficient to call Zoltan, since that also needs the number of chares we need Zoltan to partition the mesh into. The number of chares is computed in `Transporter::load()` another `reductiontarget` with necessary contributions from all `Carrier` chares. Only when both are ready can we call Zoltan. Do you agree?

It is, of course, possible to do individual (point-to-point) calls to replace both reductions, but they will not use Charm++'s tree-based aggregation, and they are a lot less perfomant (I tested this). However, both the total number of chares (into which we want to partition the mesh) and the fact that the mesh has been read and its centroids have been computed must be completed before we can call Zoltan.

What you write is general and obvious. In light of the above two paragraphs with specific information on the problem and its constraints on correctness, can you offer a better alternative?